tests/foreman/longrun/test_oscap.py

```diff
@@ -42,670 +44,617 @@
-                        'weekday': OSCAP_WEEKDAY['friday'].lower(),
-                        'organizations': self.config_env['org_name'],
-                    }
-                )
-                Arfreport.delete({'id': arf_report[0].get('id')})
-                for _ in range(2):
-                    vm.run('puppet agent -t 2> /dev/null')
-                updated_result = vm.run(
-                    'cat /etc/foreman_scap_client/config.yaml | grep content_path'
-                )
-                assert result != updated_result
-                assert updated_result.return_code == 0
-                # Runs the actual oscap scan on the vm/clients and
-                # uploads report to Internal Capsule.
-                vm.execute_foreman_scap_client()
-                result = Arfreport.list({'search': f'host={vm.hostname.lower()}'})
-                assert result is not None
-
-    @pytest.mark.upgrade
-    @pytest.mark.libvirt_content_host
-    @pytest.mark.tier4
-    def test_positive_oscap_run_with_tailoring_file_and_capsule(self):
-        """End-to-End Oscap run with tailoring files and default capsule via puppet
-
-        :id: 346946ad-4f62-400e-9390-81817006048c
-
-        :setup: scap content, scap policy, tailoring file, host group
-
-        :steps:
-
-            1. Create a valid scap content
-            2. Upload a valid tailoring file
-            3. Create a scap policy
-            4. Associate scap content with it's tailoring file
-            5. Associate the policy with a hostgroup
-            6. Provision a host using the hostgroup
-            7. Puppet should configure and fetch the scap content
-               and tailoring file
-
-        :expectedresults: ARF report should be sent to satellite reflecting
-                         the changes done via tailoring files
-
-        :BZ: 1722475
-
-        :CaseImportance: Critical
-        """
-        if settings.rhel7_repo is None:
-            self.skipTest('Missing configuration for rhel7_repo')
-        rhel7_repo = settings.rhel7_repo
-        hgrp7_name = gen_string('alpha')
-        policy_values = {
-            'content': self.rhel7_content,
-            'hgrp': hgrp7_name,
-            'policy': gen_string('alpha'),
-            'profile': OSCAP_PROFILE['security7'],
+@pytest.fixture(scope='module', autouse=True)
+def update_scap_content(module_org):
+    """ Update default scap content"""
+    for content in rhel8_content, rhel7_content, rhel6_content:
+        content = Scapcontent.info({'title': content}, output_format='json')
+        organization_ids = [content_org['id'] for content_org in content.get('organizations', [])]
+        organization_ids.append(module_org.id)
+        Scapcontent.update({'title': content['title'], 'organization-ids': organization_ids})
+
+
+@pytest.fixture(scope='module', autouse=True)
+def scap_setup(module_org, default_proxy, content_view, lifecycle_env, puppet_env):
```

Comment:
The structure above looks a lot better! But I think you have `autouse=True` and don't need it, in your case the tests are using `scap_setup`, which in turn will call all of the listed fixtures without the need for autouse.

I would add this fixture to tests that need it but aren't currently listing it instead of using autouse here.

Reply:
correct. in general, autouse should be for fixtures that aren't direct dependencies of other tests but need to run anyway.

Reply:
added it there by mistake, it was not intended. :P

Reply:
updated pr